python/tvm/auto_scheduler/task_scheduler.py

```diff
@@ -0,0 +1,452 @@
+        self.task_costs_history = [[] for _ in range(len(self.tasks))]
+
+        # best_costs[i] saves the best latency of task i
+        self.best_costs = 1e10 * np.ones(len(self.tasks))
+
+        self.tune_option = self.measurer = self.search_policies = self.ct = self.tic = None
+        self.num_measures_per_round = None
+        self.dead_tasks = set()
+        self.sequential_now_task_idx = 0
+        self.sequential_now_task_begin_ct = 0
+
+        assert len(tasks) != 0, "No tasks"
+
+        # Build similarity group
+        self.task_tags = []
+        self.tag_to_group_id = {}
+        self.group_task_ids = []
+        self.flop_cts = []
+        for i, task in enumerate(self.tasks):
+            tag = derive_similarity_tag(task.compute_dag)
+            self.task_tags.append(tag)
+            self.flop_cts.append(task.compute_dag.flop_ct)
+            if tag == "":
+                continue
+
+            if tag not in self.tag_to_group_id:
+                self.tag_to_group_id[tag] = len(self.tag_to_group_id)
+                self.group_task_ids.append([])
+            self.group_task_ids[self.tag_to_group_id[tag]].append(i)
+
+    def tune(self, tune_option, search_policy="default"):
+        """Tune a batch of tasks together.
+
+        Parameters
+        ----------
+        tune_option: TuningOptions
+            The options of tuning
+        search_policy: : Union[str, List[SearchPolicy]]
+            The list of search policies.
+            If it is str.
+            "sketch.xgb" for SketchPolicy + XGBModel
+            "sketch.random" for SketchPolicy + RandomModel
+        """
+        # init members
+        self.tune_option = tune_option
+        self.measurer = ProgramMeasurer(
+            tune_option.builder,
+            tune_option.runner,
+            tune_option.measure_callbacks,
+            tune_option.verbose,
+        )
+        self.ct = 0
+        self.tic = time.time()
+        self.sequential_now_task_idx = 0
+        self.sequential_now_task_begin_ct = 0
+        # reset num_measures_per_round to make sure every task is tuned at least once
+        self.num_measures_per_round = min(
+            tune_option.num_measures_per_round, tune_option.num_measure_trials // len(self.tasks)
+        )
+        assert self.num_measures_per_round > 0, "num_measure_trials is too small"
+
+        # restore the status of the task scheduler from a log file
+        self.restore_status(self.load_log_file, self.num_measures_per_round)
+
+        # make one search policy for one task
+        self.search_policies = make_search_policies(
+            search_policy,
+            self.tasks,
+            self.num_measures_per_round,
+            self.load_model_file,
+            self.load_log_file,
+        )
+        for i in range(len(self.tasks)):
+            search_policy = self.search_policies[i]
+            search_policy.set_verbose(tune_option.verbose)
+            # todo(merrymercy): call presearch callbacks?
```

Comment:
Since currently we have "verbose" and "init_search_callbacks" in SketchPolicy's construction function as input parameters, it's better to just move these to `make_search_policies()`.
The `set_verbose`, `set_task` and `run_callbacks` in our original repo are no longer needed.